python_bindings/caramel/__init__.py

```diff
@@ -0,0 +1,75 @@
+from ._caramel import *
+
+
+def CSF(keys, values, max_to_validate=None):
+    """
+    Constructs a CSF, automatically inferring the correct CSF backend.
+
+    Arguments:
+        keys: List of hashable keys.
+        values: List of values to use in the CSF.
+        max_to_validate: If provided, only the first "max_to_validate" values
```

Comment:
cool!

Reply:
Thanks - I like this design, but I can already anticipate all kinds of bugs in the automatic CSF dispatcher. Choosing the wrong CSF backend, etc. We'll see whether this is a bad API lol

Reply:
It may lead to a couple bugs initially but I like the direction this is going! Push button is king.